### PR TITLE
[MRG] Add a tracing function to logging

### PR DESCRIPTION
**Pull request by luizirber**
I've been using the `trace` function for generating sunburst plots of SBT searches in #1201, but it is more useful generally before that PR is merged.

This PR also fixes the formatting of `sbt.print()`, preparing it to the near future (when the internal node names won't be necessarily `internal.0` anymore), and also avoiding printing unoccupied positions in the tree.

## Checklist

- [ ] Is it mergeable?
- [ ] `make test` Did it pass the tests?
- [ ] `make coverage` Is the new code covered?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by luizirber** [comment]
Maybe TODO: figure out how to test both `sbt.print()` and the `logging.trace`. `print` is output parsing testing, but `trace` is harder because it needs to set an env var for where to save the trace data...

File: sourmash/sbt.py [hunk 1]
```diff
@@ -327,9 +328,10 @@ def find(self, search_fn, *args, **kwargs):
                     elif isinstance(node_g, Node):
                         if kwargs.get('dfs', True):  # defaults search to dfs
                             for c in self.children(node_p):
-                                queue.insert(0, c.pos)
+                                if c.node:
```
Comment (ctb) [inline sourmash/sbt.py:331]: could you explain/comment these changes, pls?

**Review by ctb — CHANGES_REQUESTED** [review]
Looks good, but I'd suggest a few more things; see comment in particular. One  thing that would be helpful to add is some example output, so I can see what this PR actually does (or intends to do ;).

File: sourmash/sbt.py (hunk 1 above)
Comment (luizirber) [reply sourmash/sbt.py:331]: the node was being inserted in the search queue even if the node data was empty, which is a small difference but avoids checking it in the next iteration. Now it is only inserted in the queue if the node actually exists (and is not `None`)

Comment (ctb) [reply sourmash/sbt.py:331]: 👍

**Comment by luizirber** [comment]
There are better approaches to do this, and SBTs are not my priority nowadays

**Pull request closed without merging**